lib/xrp/XrpRpc.js

```diff
@@ -0,0 +1,186 @@
+    if (passphrase && !secret) {
+      secret = passphrase;
+    }
+    if (!secret) {
+      throw new Error('Secret not provided');
+    }
+    let sourceAddress = this.rpc.deriveAddress(this.rpc.deriveKeypair(secret).publicKey);
+    const payment = {
+      source: {
+        address: sourceAddress,
+        tag: tag || undefined,
+        maxAmount: {
+          value: amount.toString(),
+          currency: 'XRP'
+        }
+      },
+      destination: {
+        address: address,
+        tag: tag || undefined,
+        amount: {
+          value: amount.toString(),
+          currency: 'XRP'
+        }
+      },
+      invoiceID: invoiceID || undefined,
+    };
+    if (!alreadyConnected) {
+      await this.rpc.connect();
+    }
+    return this.rpc.preparePayment(sourceAddress, payment).then(async (prepared) => {
+      const { signedTransaction } = this.rpc.sign(prepared.txJSON, secret);
+      return this.rpc.submit(signedTransaction).then(async (response) => {
+        /* Must ensure transaction confirms / validates using getTransaction later on.
+         * Pending transactions will have status: 'success' & undefined meta field.
+         * Validated (confirmed) transactions will have meta & validated: true.
+         */
+        if (!alreadyConnected) {
+          await this.rpc.disconnect();
+        }
+        if (response.resultCode !== 'tesSUCCESS') {
+          return new Error('Failed to submit transaction.');
+        }
+        return response.tx_json.hash;
+      });
+    });
+  }
+
+  async unlockAndSendToAddressMany({ payToArray, passphrase}) {
+    if (passphrase === undefined) {
+      passphrase = await passwordPromptAsync('> ');
+    }
+
+    const resultArray = [];
+    await this.rpc.connect();
+    for (const payment of payToArray) {
+      const { address, amount, id } = payment;
+      const emitData = { address, amount, id };
+      this.emitter.emit('attempt', emitData);
+      try {
+        const txid = await this.sendToAddress({ address, amount, passphrase }, true);
+        emitData.txid = txid;
+        resultArray.push(emitData);
+        this.emitter.emit('success', emitData);
+
+        //do not await confirmations,
+      } catch(e) {
+        emitData.error = e;
+        resultArray.push(emitData);
+        this.emitter.emit('failure', emitData);
+      }
+    }
+    return resultArray;
+  }
+
+  async estimateFee() {
+    return await this.asyncCall('getFee');
+  }
+
+  async getBalance({ address }) {
+    let balance =  await this.asyncCall('getBalances', this.address || address);
+    return balance.find((b) => (b.currency === 'XRP')).value;
+  }
+
+  async getBestBlockHash() {
+    let tip = await this.getTip();
+    return tip.hash;
+  }
+
+  async getTransaction({ txid }) {
+    const tx = await this.asyncRequest('tx', {transaction:txid});
+    if (!tx) {
+      return null;
+    }
+    return tx;
```

Comment:
Fixed

Reply:
If you invert this, and move the confirmations math here, instead of calling this.getConfirmations, and for this.getConfirmations you can just return 

`this.getTransaction(txid).confirmations`

Right now it's making two requests for get tx each time.

Reply:
Fixed